contracts/Dispatcher.sol

```diff
@@ -541,6 +596,84 @@ contract Dispatcher is IbcDispatcher, Ownable {
         emit Acknowledgement(address(receiver), packet.src.channelId, packet.sequence);
     }
 
+    /**
+     * @notice Handle the incentivized acknowledgement of an IBC packet by the counterparty
+     * @dev Verifies the given proof and calls the `onAcknowledgementPacket` function on the given `receiver` contract,
+     *   ie. the IBC dApp.
+     *   Prerequisite: the original packet is committed and not ack'ed or timed out yet.
+     * @param receiver The dApp contract that should handle the app-level packet acknowledgement
+     * @param packet The original IbcPacket data sent by the dApp
+     * @param incentivizedAck The incentivized acknowledgement from counterparty chain, where the relayer is the payee address on behalf of the forward relayer that delivered the packet
+     * @param proof The membership proof to verify the packet acknowledgement committed on Polymer chain
+     */
+    function incentivizedAcknowledgement(
```

Comment:
Why do we have a separate method for incentivised ack? Who's gonna call the non incentivised one?

Reply:
Depending on if the channel is incentivized or not, the ack payload is different. So relayers must call the correct one per channel type. Separate entries is for gas optimization; otherwise we have to unmarshal bytes to two ack types